src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier extends Figure {
+    private static final int FIGURES_NUMBER = 5;
+    private static final int MAX_RANDOM_NUMBER = 20;
+    private static final int DEFAULT_RADIUS = 10;
+
+    public Figure getRandomFigure() {
+        int indexOfFigures = new Random().nextInt(FIGURES_NUMBER);
+
+        ColorSupplier colorSupplier = new ColorSupplier();
```

Comment:
let's make Random and ColorSupplier as class level variables, not to create new instances every time we call getRandomFigure() method.